### PR TITLE
ESSOptimizer: Fix bug in go-beyond

### PR DESCRIPTION
**Pull request by dweindl**
Fixes some issues related to copies vs views that stopped the go-beyond search prematurely after the first successful step.

### CODE REVIEW

File: pypesto/optimize/ess/ess.py [hunk 1]
```diff
@@ -583,9 +583,9 @@ def _go_beyond(self, x_best_children, fx_best_children):
                 continue
 
             # offspring is better than parent
-            x_parent = self.refset.x[i]
+            x_parent = self.refset.x[i].copy()
```
Comment (PaulJonasJost) [inline pypesto/optimize/ess/ess.py:586]: Do i see correctly that the problem was that `x_parent/child` are edited further below and thus edit the reset itself?

File: pypesto/optimize/ess/ess.py [hunk 2]
```diff
@@ -560,7 +560,7 @@ def _do_local_search(
     def _maybe_update_global_best(self, x, fx):
         """Update the global best value if the provided value is better."""
         if fx < self.fx_best:
-            self.x_best = x[:]
+            self.x_best[:] = x
```
Comment (PaulJonasJost) [inline pypesto/optimize/ess/ess.py:563]: how is this related to the go-beyond algorithm?

**Review by PaulJonasJost — APPROVED** [review]
Looks good, thanks. I added some questions for my own clarification 🙏🏼

File: pypesto/optimize/ess/ess.py (hunk 2 above)
Comment (dweindl) [reply pypesto/optimize/ess/ess.py:563]: It's not. But it's another potential copying issue.

File: pypesto/optimize/ess/ess.py (hunk 1 above)
Comment (dweindl) [reply pypesto/optimize/ess/ess.py:586]: The problem was not so much editing the refset per se, but that in the second inner iteration, `x_parent ` would become identical to `x_child`.
